dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,851 @@
+        try {
+
+            Identifier identifier = APILocator.getIdentifierAPI()
+                .find(id1);
+            Contentlet contentlet = null;
+            if (CacheLocator.getVeloctyResourceCache()
+                .isMiss(filePath)) {
+                if (LanguageWebAPI.canDefaultContentToDefaultLanguage()) {
+                    LanguageAPI langAPI = APILocator.getLanguageAPI();
+                    language = Long.toString(langAPI.getDefaultLanguage()
+                        .getId());
+                } else {
+                    throw new ResourceNotFoundException("Contentlet is a miss in the cache");
+                }
+            }
+
+            try {
+                contentlet = APILocator.getContentletAPI()
+                    .findContentletByIdentifier(identifier.getId(), mode.showLive, new Long(language), APILocator.getUserAPI()
+                        .getSystemUser(), true);
+            } catch (DotContentletStateException e) {
+                contentlet = null;
+            }
+
+            if (contentlet == null || !InodeUtils.isSet(contentlet.getInode()) || contentlet.isArchived()) {
+
+                LanguageAPI langAPI = APILocator.getLanguageAPI();
+                long lid = langAPI.getDefaultLanguage()
+                    .getId();
+                if (lid != Long.parseLong(language)) {
+                    Contentlet cc = APILocator.getContentletAPI()
+                        .findContentletByIdentifier(identifier.getId(), mode.showLive, lid, APILocator.getUserAPI()
+                            .getSystemUser(), true);
+                    if (cc != null && UtilMethods.isSet(cc.getInode()) && !cc.isArchived()
+                            && LanguageWebAPI.canApplyToAllLanguages(cc)) {
+                        contentlet = cc;
+                    } else {
+                        CacheLocator.getVeloctyResourceCache()
+                            .addMiss(filePath);
+                        throw new ResourceNotFoundException("Contentlet Velocity file not found:" + filePath);
+                    }
+                }
+            }
+
+            Logger.debug(this, "DotResourceLoader:\tWriting out contentlet inode = " + contentlet.getInode());
+            return buildVelocity(contentlet, identifier, mode, filePath);
+
+        } catch (Exception e) {
+            throw new ResourceNotFoundException("Contentlet Velocity file not found:" + filePath);
+        }
+    }
+
+
+
+
+    @Override
+    public void invalidate(Object obj, PageMode mode) {
+        Contentlet asset = (Contentlet) obj;
+        CacheLocator.getContentletCache()
+            .remove(asset.getInode());
+
+        String folderPath = mode.name() + java.io.File.separator;
+        String velocityRootPath = VelocityUtil.getVelocityRootPath();
+        velocityRootPath += java.io.File.separator;
+
+        Set<Long> langs = new HashSet<Long>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)